### ISSUE TITLE
Add support for hydra configs

### ISSUE THREAD

**Issue by gorarakelyan** [aimhubio/aim#404]
It would be great to support logging hydra(https://github.com/facebookresearch/hydra) configs directly without preliminary conversion to python dicts. For example `aim.set_params(hydra_cfg)`.

**Comment by felixkreuk** [comment]
Would the `set_params` method of `Session` class be a good spot for this?
Thinking about it, it would introduce an OmegaConf dependency (hydra uses dictConfig), is that something you want?

**Comment by gorarakelyan** [comment]
Yea, probably OmegaConf is the way to go. 

re `set_params`: For logging configs set_params is used currently, do you think there is a better way/interface of integration with hydra?

Thanks for sharing details. Really appreciate your help 🙏

**Comment by haritsahm** [comment]
I got this error when using pytorch-lightning+hydra and aim. So aim will log the params but DictConfig from OmegaConf is not supported. How do I resolve this?

```
Traceback (most recent call last):
  File "/opt/conda/lib/python3.8/site-packages/hydra/_internal/utils.py", line 252, in run_and_report
    assert mdl is not None
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "run.py", line 34, in <module>
    main()
  File "/opt/conda/lib/python3.8/site-packages/hydra/main.py", line 48, in decorated_main
    _run_hydra(
  File "/opt/conda/lib/python3.8/site-packages/hydra/_internal/utils.py", line 377, in _run_hydra
    run_and_report(
  File "/opt/conda/lib/python3.8/site-packages/hydra/_internal/utils.py", line 294, in run_and_report
    raise ex
  File "/opt/conda/lib/python3.8/site-packages/hydra/_internal/utils.py", line 211, in run_and_report
    return func()
  File "/opt/conda/lib/python3.8/site-packages/hydra/_internal/utils.py", line 378, in <lambda>
    lambda: hydra.run(
  File "/opt/conda/lib/python3.8/site-packages/hydra/_internal/hydra.py", line 111, in run
    _ = ret.return_value
  File "/opt/conda/lib/python3.8/site-packages/hydra/core/utils.py", line 233, in return_value
    raise self._return_value
  File "/opt/conda/lib/python3.8/site-packages/hydra/core/utils.py", line 160, in run_job
    ret.return_value = task_function(task_cfg)
  File "run.py", line 30, in main
    return train(config)
  File "/home/haritsahm/Documents/projects/deeplearning/ball-detection-pytorch/src/train.py", line 66, in train
    utils.log_hyperparameters(
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/utilities/distributed.py", line 49, in wrapped_fn
    return fn(*args, **kwargs)
  File "/home/haritsahm/Documents/projects/deeplearning/ball-detection-pytorch/src/utils/utils.py", line 153, in log_hyperparameters
    trainer.logger.log_hyperparams(hparams)
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/loggers/base.py", line 411, in log_hyperparams
    logger.log_hyperparams(params)
  File "/opt/conda/lib/python3.8/site-packages/pytorch_lightning/utilities/distributed.py", line 49, in wrapped_fn
    return fn(*args, **kwargs)
  File "/opt/conda/lib/python3.8/site-packages/aim/sdk/adapters/pytorch_lightning.py", line 63, in log_hyperparams
    self.experiment['hparams'] = params
  File "/opt/conda/lib/python3.8/site-packages/aim/sdk/run.py", line 329, in __setitem__
    self.meta_run_attrs_tree[key] = val
  File "/opt/conda/lib/python3.8/site-packages/aim/storage/containertreeview.py", line 101, in __setitem__
    for key, val in treeutils.encode_tree(value):
  File "aim/storage/treeutils_.pyx", line 165, in encode_paths_vals
  File "aim/storage/treeutils_.pyx", line 48, in unfold_tree
  File "aim/storage/treeutils_.pyx", line 58, in unfold_tree
TypeError: Not supported value `{'_target_': 'pytorch_lightning.Trainer', 'gpus': 1, 'min_epochs': 1, 'max_epochs': 10, 'progress_bar_refresh_rate': 5, 'resume_from_checkpoint': None, 'weights_summary': 'full', 'num_sanity_val_steps': 1, 'gradient_clip_val': 0.5}` of type `<class 'omegaconf.dictconfig.DictConfig'>`.
```

**Comment by alberttorosyan** [comment]
Hi @haritsahm. Thanks for reporting this issue.
Currently `aim` storage does not support dict-like configs.
However to fix this specific issue we need to change the `pytorch` lightning adapter to do the config conversion to python dict prior to setting it to Run.
I've opened a separate issue #1175 for that matter.

**Comment by haritsahm** [comment]
@alberttorosyan thank you for the reply. I will wait for the new fix

**Comment by gorarakelyan** [comment]
@haritsahm @felixkreuk Hydra config support has been shipped with Aim v3.3. Please feel free to reopen this issue in case of any unexpected behavior 🙌
